src/common/s3_adapter.cpp

```diff
@@ -332,6 +335,11 @@ int S3Adapter::PutObject(const Aws::String &key, const std::string &data) {
 */
 
 void S3Adapter::PutObjectAsync(std::shared_ptr<PutObjectAsyncContext> context) {
+    asyncRequestTP_.Enqueue(&S3Adapter::PutObjectAsyncHelper, this, context);
```

Comment:
can cause excessive memory usage？

Reply:
I have the same question

Reply:
yes, i will change it, seems that just put callbacks into queue is enough.